runtime/plaid/src/apis/gcp/google_docs.rs

```diff
@@ -0,0 +1,639 @@
+use std::collections::{HashMap, HashSet};
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use plaid_stl::gcp::google_docs::{
+    CreateDocFromMarkdownInput, CreateDocFromMarkdownOutput, CreateSheetFromCsvInput,
+};
+use pulldown_cmark::{html, Options, Parser};
+use reqwest::{multipart, Client};
+use serde::Deserialize;
+use serde_json::{json, Value};
+use tera::{Context, Tera};
+use thiserror::Error;
+use tokio::sync::Mutex;
+
+use crate::apis::{AccessScope, ApiError};
+use crate::loader::PlaidModule;
+
+/// Defines configuration for the Google Docs API
+#[derive(Deserialize)]
+pub struct GoogleDocsConfig {
+    /// Google OAuth Client ID
+    client_id: String,
+    /// Google OAuth Client Secret
+    client_secret: String,
+    /// Google OAuth refresh token, used to obtain valid OAuth Access Token
+    refresh_token: String,
+    /// Configured writers - maps a folder ID to a list of rules that are allowed to READ or WRITE files
+    rw: HashMap<String, HashSet<String>>,
+    /// Configured readers - maps a folder ID to a list of rules that are allowed to READ files
+    r: HashMap<String, HashSet<String>>,
+}
+
+/// Represents the Google Docs API client
+pub struct GoogleDocs {
+    /// Inner HTTP client used to make requests to Google APIs
+    client: Client,
+    /// Google OAuth Client ID
+    client_id: String,
+    /// Google OAuth Client Secret
+    client_secret: String,
+    /// Google OAuth refresh token for plaid's google account
+    refresh_token: String,
+    /// Cached Google OAuth Access Token (Token, Expiry)
+    access_token: Mutex<Option<(String, Instant)>>,
+    /// Configured writers - maps a folder ID to a list of rules that are allowed to READ or WRITE files
+    rw: HashMap<String, HashSet<String>>,
+    /// Configured readers - maps a folder ID to a list of rules that are allowed to READ files
+    r: HashMap<String, HashSet<String>>,
+}
+
+#[derive(Error, Debug)]
+pub enum GoogleDocsError {
+    #[error("HTTP error: {0}")]
+    Reqwest(#[from] reqwest::Error),
+    #[error("Missing field in response: {0}")]
+    MissingField(&'static str),
+    #[error("OAuth error: {description}")]
+    OAuth { description: String },
+    #[error("Google API error: {0}")]
+    GoogleApi(String),
+    #[error("Template error: {0}")]
+    Template(String),
+}
+
+// Google OAuth Token URL
+const TOKEN_URL: &str = "https://oauth2.googleapis.com/token";
+// Google Drive Multipart Upload URL
+const DRIVE_UPLOAD_URL: &str =
+    "https://www.googleapis.com/upload/drive/v3/files?uploadType=multipart";
+
+impl GoogleDocs {
+    pub fn new(config: GoogleDocsConfig) -> Self {
+        let GoogleDocsConfig {
+            client_id,
+            client_secret,
+            refresh_token,
+            r,
+            rw,
+        } = config;
```

Comment:
You could also just store the config inside the `GoogleDocs` struct which will then keep them in sync if you ever need to add anything